lab_2/main.py

```diff
@@ -49,103 +58,37 @@ def get_language_features(language_profiles: dict) -> list or None:
-    pass
-
-
-# 6
-def calculate_distance(unknown_text_vector: list, known_text_vector: list) -> float or None:
-    """
-    Calculates distance between two vectors using euclid metric
-    :param unknown_text_vector: vector for unknown text
-    :param known_text_vector: vector for known text
-    """
-    pass
-
-
-def predict_language_score(unknown_text_vector: list, known_text_vectors: list,
-                           language_labels: list) -> [str, int] or None:
-    """
-    Predicts unknown text label and its distance to the closest known text
-    :param unknown_text_vector: vector for unknown text
-    :param known_text_vectors: a list of vectors for known texts
-    :param language_labels: language labels for each known text
-    """
-    pass
-
-
-# 8
-def calculate_distance_manhattan(unknown_text_vector: list,
-                                 known_text_vector: list) -> float or None:
-    """
-    Calculates distance between two vectors using manhattan metric
-    :param unknown_text_vector: vector for unknown text
-    :param known_text_vector: vector for known text
-    """
-    pass
-
-
-def predict_language_knn(unknown_text_vector: list, known_text_vectors: list,
-                         language_labels: list, k=1, metric='manhattan') -> [str, int] or None:
-    """
-    Predicts unknown text label and its distance to the closest known text
-        using knn based algorithm and specific metric
-    :param unknown_text_vector: vector for unknown text
-    :param known_text_vectors: a list of vectors for known texts
-    :param language_labels: language labels for each known text
-    :param k: the number of neighbors to choose label from
-    :param metric: specific metric to use while calculating distance
-    """
-    pass
-
-
-# 10 implementation
-def get_sparse_vector(original_text: list, language_profiles: dict) -> list or None:
-    """
-    Builds a sparse vector representation of a given text
-        using dictionary with language profiles
-    :param original_text: any tokenized text
-    :param language_profiles: a dictionary of dictionaries - language profiles
-    """
-    pass
-
-
-def calculate_distance_sparse(unknown_text_vector: list,
-                              known_text_vector: list) -> float or None:
-    """
-    Calculates distance between two vectors using euclid metric
-    :param unknown_text_vector: sparse vector for unknown text
-    :param known_text_vector: sparse vector for known text
-    """
-    pass
-
+    if not isinstance(original_text, list):
+        return None
+    for elem in original_text:
+        if not isinstance(elem, str):
+            return None
+    features = get_language_features(language_profiles)
+    # original_text = set(original_text)
+    text_vector = []
+    for feature in features:
+        if feature not in original_text:
+            text_vector.append(0)
+        else:
+            text_vector.append(original_text.count(feature) / len(original_text))
```

Comment:
looks like you did this here `get_freq_dict`

Reply:
get freq dict returns dict but we need a list